frontend/src/components/common/HeaderWithButton.tsx

```diff
@@ -6,11 +6,12 @@ import { PlusOutlineIcon } from "../../assets/icons";
 
 type HeaderWithButtonProps = {
   buttonIcon?: React.ReactElement;
-  buttonText: string;
+  buttonText?: string;
   onClick?: () => void;
   showButton?: boolean;
   targetRoute?: string;
   title?: string;
+  buttonComponent?: React.ReactElement;
```

Comment:
LOL I had actually been thinking you'd add an `onClick` handler instead of a component prop but this works well too. Just make sure the styling matches other pages (e.g., top margin)

Reply:
I thought about it and decided that onClick would be cleaner, so i made the changes! Let me know what you think